capabilityquerier/pkg/capabilityquerier/capabilityquerier.go

```diff
@@ -246,80 +248,65 @@ func requestCapabilityStatementAndSmartOnFhir(ctx context.Context, fhirURL strin
 		req.Header.Set("fhirVersion", message.RequestedFhirVersion)
 	}
 
-	firstMIME := fhir3PlusJSONMIMEType
-	randomMimeIdx := 0
-
 	// If there are mime types saved in the database for this URL
-	if endptType == metadata && len(message.MIMETypes) > 0 {
-		// Choose a random mime type in the list if there's more than one
-		if len(message.MIMETypes) == 2 {
-			rand.Seed(time.Now().UnixNano())
-			randomMimeIdx = rand.Intn(2)
-			firstMIME = message.MIMETypes[randomMimeIdx]
-		} else {
-			firstMIME = message.MIMETypes[randomMimeIdx]
-		}
-		httpResponseCode, tlsVersion, mimeTypeWorked, capResp, responseTime, err = requestWithMimeType(req, firstMIME, client)
-		if err != nil {
-			return err
-		}
-	} else if endptType == wellknown && len(message.MIMETypes) > 0 {
-		firstMIME = message.MIMETypes[0]
-		httpResponseCode, _, _, capResp, _, err = requestWithMimeType(req, firstMIME, client)
-		if err != nil {
-			return err
-		}
-	} else {
-		httpResponseCode, tlsVersion, mimeTypeWorked, capResp, responseTime, err = requestWithMimeType(req, fhir3PlusJSONMIMEType, client)
-		if err != nil {
+	if len(message.MIMETypes) == 1 {
+		// Only one MIME type saved
+		savedMIME := message.MIMETypes[0]
+		httpResponseCode, tlsVersion, mimeTypeWorked, capResp, responseTime, httpErr = requestWithMimeType(req, savedMIME, client)
+		if httpErr != nil && httpResponseCode != 0 {
```

Comment:
I just want to make sure I understand this, so if there is an error reading the capability statement, then it returns an error, but if we get a valid httpResponseCode that's an error, or we get a timeout error, then we continue with the rest of the process? Is there a reason why this is different than the others?

Reply:
also could you add some comments through this section to explain the process? I think it might be confusing to anyone that isn't familiar with what we're trying to do

Reply:
Hmm I am not sure why it is the case that this even checks the capability statement here, as it seems the only reason it does is to check if it can be read or not. Not sure if removing this would mess up something somewhere else, but I am not too sure why this needs to be here